listenbrainz/db/release.py

```diff
@@ -0,0 +1,101 @@
+from typing import Iterable
+
+from psycopg2.extras import execute_values
+from listenbrainz.db.musicbrainz_entity import resolve_redirect_mbids
+
+
+def load_releases_from_mbids(ts_curs, mbids: Iterable[str]) -> dict:
+    """ Given a list of mbids return a map with mbid as key and the release info as value.
+
+
+    TODO: Ideally, we should always resolve redirects but that makes testing some LB features difficult
+        because we don't have a MB db in the test environment yet. Once, we do we can merge the two versions.
+    """
+    if not mbids:
+        return {}
+
+    query = """
```

Comment:
We are coming up with an artist cache for the entity pages which is much more efficient and complete in #2667. I think we can just use that instead of creating this new endpoint.

Reply:
Since we're fetching the release data as well, so we don't have the required data in the artist cache. So we'd need to fetch the data from mb_metadata_cache

Reply:
We have release group data in artist cache, we should show those instead of releases unless absolutely necessary.

Reply:
Example: https://test-api.listenbrainz.org/1/metadata/artist/?artist_mbids=20244d07-534f-4eff-b4d4-930878889970&inc=release_group

Reply:
Makes sense, it should work for our needs

Reply:
Instead can we use the mb_release_group_cache?

Reply:
If you have an artist mbid and need the artists' release groups, you need to use the artist cache. If you have release group mbids (if you have release mbids even those could be resolved to release groups), you can use the release group cache.